parity/cli/mod.rs

```diff
@@ -318,6 +318,10 @@ usage! {
 			"--db-path=[PATH]",
 			"Specify the database directory path",
 
+			ARG arg_sync_until: (Option<u64>) = None, or |c: &Config| c.parity.as_ref()?.sync_until.clone(),
+			"--sync-until=[NUM]",
+			"Stop syncing and enter offline mode when the given block has been imported.",
```

Comment:
is there a way to hide this flag from the user, I don't think we should expose to the general audience?

Reply:
If I replace lines 322 and 323 with `"",\n""` the help output looks like this (note the extra white space between the sections):

```
    --identity=[NAME]
        Specify your node's name. (default: )

    -d, --base-path=[PATH]
        Specify the base data storage path.

    --db-path=[PATH]
        Specify the database directory path

    


Convenience Options:
    --unsafe-expose
        All servers will listen on external interfaces and will be remotely accessible. It's equivalent with setting the
        following: --[ws,jsonrpc,secretstore,stratum,dapps,secretstore-http]-interface=all --*-hosts=all    This option
        is UNSAFE and should be used with great care!

    -c, --config=[CONFIG]
        Specify a configuration. CONFIG may be either a configuration file or a preset: dev, insecure, dev-insecure,
        mining, or non-standard-ports. (default: $BASE/config.toml)

    --ports-shift=[SHIFT]
        Add SHIFT to all port numbers Parity is listening on. Includes network port and all servers (HTTP JSON-RPC,
        WebSockets JSON-RPC, SecretStore). (default: 0)

Account Options:
    --fast-unlock
        Use drastically faster unlocking mode. This setting causes raw secrets to be stored unprotected in memory, so use with care.
```

So no, I don't think there is an easy way. Our CLI/config system is… peculiar. :/

OTOH do you really expect someone to `--sync-until=123` and then come file a bug about a stalled node?

Reply:
> OTOH do you really expect someone to --sync-until=123 and then come file a bug about a stalled node?

No, I'm just concerned about showing users an option in the `--help` output that they are not supposed to use.

Reply:
> No, I'm just concerned about showing users an option in the `--help` output that they are not supposed to use.

I don't have a good suggestion here.

Reply:
I think that we have to populate this benchmarking/debugging flag as a special flag for the moment, so my vote is to rename it as `--unsafe-sync-until` or `--experimental-sync-until`

Reply:
I don't mind renaming it, but I don't like `unsafe` (it's not unsafe in any way) and I don't like `experimental` either (it's not a feature we're planning on working on, there's nothing experimental about it).
How about extending the help text? Maybe "Sync until the given block has been imported then enter sleep mode. Intended for debug/benchmarking only."?

Reply:
Yep that unsafe or experimental is not the proper naming for this. Agree that _Intended for debug/benchmarking only._ also does the job.